spotify.go

```diff
@@ -124,6 +175,74 @@ func decodeError(r io.Reader) error {
 // authenticate, you can use `DefaultClient`.
 type Client struct {
 	http *http.Client
+	retryDuration time.Duration
+}
+
+func (c *Client) ExecuteOpt(req *http.Request, needsStatus int, result interface{}) (err error) {
```

Comment:
`needsStatus` is a secondary success code (which can be set to 0 to be ignored). In most cases we only check whether we got OK. In several of the non-GET requests (often for create calls) we also checked that it returned "created":

```
github.com/zmb3/spotify$ grep ExecuteOpt -R .
./spotify.go:func (c *Client) ExecuteOpt(req *http.Request, needsStatus int, result interface{}) (err error) {
./spotify.go:	return c.ExecuteOpt(req, 0, nil)
./playlist.go:	err = c.ExecuteOpt(req, http.StatusCreated, &p)
./playlist.go:	err = c.ExecuteOpt(req, http.StatusCreated, &body)
./playlist.go:	err = c.ExecuteOpt(req, 0, &result)
./playlist.go:	err = c.ExecuteOpt(req, 0, &result)
```

I'm not sure. If the server was telling us to wait and we threw an error or did something other than wait then I think we'd be interfering. It'd be inappropriate to apply our own interpretation of what we `think` we should do.